apps/qubit/modules/actor/actions/browseAction.class.php

```diff
@@ -216,46 +227,21 @@ protected function populateAgg($name, $buckets)
 
   protected function setFilterTags($request)
   {
-    if (isset($request->repository) && ctype_digit($request->repository))
-    {
-      $this->repository = QubitRepository::getById($request->repository);
-
-      // Add repo to the user session as realm
-      if (sfConfig::get('app_enable_institutional_scoping'))
-      {
-        $this->context->user->setAttribute('search-realm', $request->repository);
-      }
-    }
-    else if (sfConfig::get('app_enable_institutional_scoping'))
-    {
-      // Remove search realm
-      $this->context->user->removeAttribute('search-realm');
-    }
+    $this->populateFilterTags($request);
 
-    if (isset($request->entityType) && ctype_digit($request->entityType))
+    // Set label for has digital object filter tag
+    if (filter_var($request->hasDigitalObject, FILTER_VALIDATE_BOOLEAN))
     {
-      $this->entityType = QubitTerm::getById($request->entityType);
+      $this->setFilterTagLabel('hasDigitalObject', $this->i18n->__('With digital objects'));
     }
-
-    if (isset($request->occupation) && ctype_digit($request->occupation))
-    {
-      $this->occupation = QubitTerm::getById($request->occupation);
-    }
-
-    if (isset($request->place) && ctype_digit($request->place))
+    else
     {
-      $this->place = QubitTerm::getById($request->place);
+      $this->setFilterTagLabel('hasDigitalObject', $this->i18n->__('Without digital objects'));
     }
 
-    if (isset($request->subject) && ctype_digit($request->subject))
-    {
-      $this->subject = QubitTerm::getById($request->subject);
-    }
-
-    if (isset($request->mediatypes) && ctype_digit($request->mediatypes))
-    {
-      $this->mediatypes = QubitTerm::getById($request->mediatypes);
-    }
+    // Set label for empty field filter tag
+    $labelText = $this->i18n->__('Empty: %1%', array('%1%' => $this->fieldOptions[$request->emptyField]));
```

Comment:
Should we check if `$request->emptyField` is set?

Reply:
Good call.